src/test/java/com/google/devtools/build/lib/exec/SpawnInputExpanderTest.java

```diff
@@ -294,15 +335,54 @@ public void testRunfilesWithTreeArtifacts() throws Exception {
     fakeCache.put(file1, FileArtifactValue.createForTesting(file1));
     fakeCache.put(file2, FileArtifactValue.createForTesting(file2));
 
-    expander.addRunfilesToInputs(
-        inputMappings, supplier, fakeCache, artifactExpander, PathFragment.EMPTY_FRAGMENT);
+    expander.addRunfilesToInputs(inputMappings, supplier, fakeCache, artifactExpander,
+        PathMapper.NOOP, PathFragment.EMPTY_FRAGMENT);
     assertThat(inputMappings).hasSize(2);
     assertThat(inputMappings)
         .containsEntry(PathFragment.create("runfiles/workspace/treeArtifact/file1"), file1);
     assertThat(inputMappings)
         .containsEntry(PathFragment.create("runfiles/workspace/treeArtifact/file2"), file2);
   }
 
+  @Test
+  public void testRunfilesWithTreeArtifacts_pathMapped() throws Exception {
+    SpecialArtifact treeArtifact = createTreeArtifact("treeArtifact");
+    assertThat(treeArtifact.isTreeArtifact()).isTrue();
+    TreeFileArtifact file1 = TreeFileArtifact.createTreeOutput(treeArtifact, "file1");
+    TreeFileArtifact file2 = TreeFileArtifact.createTreeOutput(treeArtifact, "file2");
+    FileSystemUtils.writeContentAsLatin1(file1.getPath(), "foo");
+    FileSystemUtils.writeContentAsLatin1(file2.getPath(), "bar");
+
+    Runfiles runfiles = new Runfiles.Builder("workspace").addArtifact(treeArtifact).build();
+    ArtifactExpander artifactExpander =
+        (Artifact artifact, Collection<? super Artifact> output) -> {
+          if (artifact.equals(treeArtifact)) {
+            output.addAll(Arrays.asList(file1, file2));
+          }
+        };
+    RunfilesSupplier supplier =
+        AnalysisTestUtil.createRunfilesSupplier(
+            PathFragment.create("bazel-out/k8-opt/bin/foo.runfiles"), runfiles);
+    FakeActionInputFileCache fakeCache = new FakeActionInputFileCache();
+    fakeCache.put(file1, FileArtifactValue.createForTesting(file1));
+    fakeCache.put(file2, FileArtifactValue.createForTesting(file2));
+
+    expander.addRunfilesToInputs(
+        inputMappings,
+        supplier,
+        fakeCache,
+        artifactExpander,
+        execPath -> PathFragment.create(execPath.getPathString().replace("k8-opt/", "")),
+        PathFragment.EMPTY_FRAGMENT);
```

Comment:
In order for this test to verify that we didn't just pass in the child path, we need to make a different transformation here. Maybe replacing k8-opt with a hash of the (short) path?